src/simulator/engine/d3-simulator-engine.ts

```diff
@@ -172,33 +178,38 @@ export class D3SimulatorEngine extends Emitter<D3SimulatorEvents> {
   protected _isDragging = false;
   protected _isStabilizing = false;
 
+  // These are settings provided during construction if they are specified,
+  // or during the first call of setSettings if unspecified during construction.
+  protected _initialSettings: ID3SimulatorEngineSettings | undefined;
+
   constructor(settings?: ID3SimulatorEngineSettings) {
     super();
 
-    this.linkForce = forceLink<ISimulationNode, SimulationLinkDatum<ISimulationNode>>(this._edges).id(
-      (node) => node.id,
-    );
-    this.simulation = forceSimulation(this._nodes).force('link', this.linkForce).stop();
-
-    this.settings = Object.assign(copyObject(DEFAULT_SETTINGS), settings);
-    this.initSimulation(this.settings);
-
-    this.simulation.on('tick', () => {
-      this.emit(D3SimulatorEngineEventType.TICK, { nodes: this._nodes, edges: this._edges });
-    });
+    if (settings !== undefined) {
+      this._initialSettings = Object.assign(copyObject(DEFAULT_SETTINGS), settings);
+    }
 
-    this.simulation.on('end', () => {
-      this._isDragging = false;
-      this._isStabilizing = false;
-      this.emit(D3SimulatorEngineEventType.END, { nodes: this._nodes, edges: this._edges });
-    });
+    // TODO(dlozic): Question: TLastre, settings! are initialized in here, so they are guaranteed
+    // to be defined, hence the (!). This isn't very readable though, but if not like this there would be
+    // code duplication.
+    this.reset();
   }
 
   getSettings(): ID3SimulatorEngineSettings {
     return copyObject(this.settings);
   }
 
+  /**
+   * Applies the specified settings to the D3 simulator engine.
+   *
+   * @param {ID3SimulatorEngineSettingsUpdate} settings Partial D3 simulator engine settings (any property of settings)
+   */
   setSettings(settings: ID3SimulatorEngineSettingsUpdate) {
+    if (!this._initialSettings) {
+      this._initialSettings = Object.assign(copyObject(DEFAULT_SETTINGS), settings);
+    }
+
+    // TODO(dlozic): Question: (from line 166, 167) is this then necessary? Why not simple assign?
```

Comment:
I think you are right. It should be double-checked.

Reply:
<just check what happens if value is null>